egs2/dns_icassp22/enh1/local/download_dns4_dataset.sh

```diff
@@ -0,0 +1,222 @@
+    # clean_fullband/datasets_fullband.clean_fullband.spanish_speech_002_NA_NA.tar.bz2
+    # clean_fullband/datasets_fullband.clean_fullband.spanish_speech_003_NA_NA.tar.bz2
+    # clean_fullband/datasets_fullband.clean_fullband.spanish_speech_004_NA_NA.tar.bz2
+    # clean_fullband/datasets_fullband.clean_fullband.spanish_speech_005_NA_NA.tar.bz2
+    # clean_fullband/datasets_fullband.clean_fullband.spanish_speech_006_NA_NA.tar.bz2
+    # clean_fullband/datasets_fullband.clean_fullband.spanish_speech_007_NA_NA.tar.bz2
+    # clean_fullband/datasets_fullband.clean_fullband.spanish_speech_008_NA_NA.tar.bz2
+
+    # clean_fullband/datasets_fullband.clean_fullband.vctk_wav48_silence_trimmed_000.tar.bz2
+    # clean_fullband/datasets_fullband.clean_fullband.vctk_wav48_silence_trimmed_001.tar.bz2
+    # clean_fullband/datasets_fullband.clean_fullband.vctk_wav48_silence_trimmed_002.tar.bz2
+    # clean_fullband/datasets_fullband.clean_fullband.vctk_wav48_silence_trimmed_003.tar.bz2
+    # clean_fullband/datasets_fullband.clean_fullband.vctk_wav48_silence_trimmed_004.tar.bz2
+
+    noise_fullband/datasets_fullband.noise_fullband.audioset_000.tar.bz2
+    noise_fullband/datasets_fullband.noise_fullband.audioset_001.tar.bz2
+    noise_fullband/datasets_fullband.noise_fullband.audioset_002.tar.bz2
+    noise_fullband/datasets_fullband.noise_fullband.audioset_003.tar.bz2
+    noise_fullband/datasets_fullband.noise_fullband.audioset_004.tar.bz2
+    noise_fullband/datasets_fullband.noise_fullband.audioset_005.tar.bz2
+    noise_fullband/datasets_fullband.noise_fullband.audioset_006.tar.bz2
+
+    noise_fullband/datasets_fullband.noise_fullband.freesound_000.tar.bz2
+    noise_fullband/datasets_fullband.noise_fullband.freesound_001.tar.bz2
+
+    # datasets_fullband.dev_testset_000.tar.bz2
+    # datasets_fullband.impulse_responses_000.tar.bz2
+)
+
+###############################################################
+
+AZURE_URL="https://dns4public.blob.core.windows.net/dns4archive/datasets_fullband"
+PATH_PREFIX=$1
+
+RUN_DIR=$(pwd)
+
+mkdir -p $PATH_PREFIX
+mkdir -p $PATH_PREFIX/{clean_fullband,noise_fullband}
+
+for BLOB in ${BLOB_NAMES[@]}
+do
+    URL="$AZURE_URL/$BLOB"
+    echo "Download: $BLOB"
+
+    # DRY RUN: print HTTP response and Content-Length
+    # WITHOUT downloading the files
+    # curl -s -I "$URL" | head -n 2
+
+    wget "$URL" -O ${PATH_PREFIX}/$BLOB -c
+    if [[ $(dirname ${PATH_PREFIX}/$BLOB) != $PATH_PREFIX ]]; then
+      cd $(dirname ${PATH_PREFIX}/$BLOB)
+      cd ../
+      mv $BLOB ./
+    fi
+    tar -xjf $(basename $BLOB)
+done
+
+# delete folders created only to prevent path errors
+rmdir $PATH_PREFIX/clean_fullband
+rmdir $PATH_PREFIX/noise_fullband
+
+# get noisy wav synthesizer config file
+cd ${PATH_PREFIX}
+wget https://raw.githubusercontent.com/microsoft/DNS-Challenge/master/noisyspeech_synthesizer.cfg -O noisyspeech_synthesizer.cfg
```

Comment:
This way, does it mean that we must clone the entire repo rather than just getting the files we need?

Or, is there a way we can just fetch those specific files? Thanks a lot.

Reply:
You could clone only a specific branch by
```bash
git clone -b [branch_name] ...
```